src/test/java/org/mule/service/http/impl/service/server/HttpServiceMalformedUrlTestCase.java

```diff
@@ -0,0 +1,183 @@
+ */
+package org.mule.service.http.impl.service.server;
+
+import static java.lang.String.format;
+import static java.net.URLEncoder.encode;
+import static java.util.Collections.singletonList;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.junit.Assert.assertThat;
+import static org.hamcrest.core.StringEndsWith.endsWith;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.apache.commons.text.StringEscapeUtils.escapeHtml4;
+import static org.mule.runtime.api.metadata.MediaType.TEXT;
+import static org.mule.runtime.http.api.HttpConstants.Method.GET;
+import static org.mule.runtime.http.api.HttpConstants.Method.POST;
+import static org.mule.runtime.http.api.HttpConstants.HttpStatus.BAD_REQUEST;
+import static org.mule.runtime.http.api.HttpConstants.HttpStatus.OK;
+import static org.mule.runtime.http.api.HttpHeaders.Names.CONTENT_TYPE;
+
+import org.apache.http.StatusLine;
+import org.apache.http.client.ClientProtocolException;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.http.client.fluent.Request;
+import org.mule.runtime.core.api.util.IOUtils;
+import org.mule.runtime.http.api.HttpConstants;
+import org.mule.runtime.http.api.domain.entity.ByteArrayHttpEntity;
+import org.mule.runtime.http.api.server.PathAndMethodRequestMatcher;
+import org.mule.runtime.http.api.server.MethodRequestMatcher;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.response.HttpResponse;
+import org.mule.runtime.http.api.server.RequestHandler;
+import org.mule.service.http.impl.functional.server.AbstractHttpServerTestCase;
+import org.mule.service.http.impl.service.util.SocketRequester;
+
+public class HttpServiceMalformedUrlTestCase extends AbstractHttpServerTestCase {
+
+  private static final String LINE_SEPARATOR = System.lineSeparator();
+  private static final String MALFORMED = "/api/ping%";
+  private static final String WILDCARD = "/*";
+
+  public static final String MALFORMED_SCRIPT = "<script></script>%";
+  public static final String ENCODED_SPACE = "test/foo 1 %";
+  public static final String ENCODED_HASHTAG = "test/foo 1 #";
+  public static final String ENCODED_PERCENT2 = "test/%24";
+
+  public HttpServiceMalformedUrlTestCase(String serviceToLoad) {
+    super(serviceToLoad);
+  }
+
+  @Override
+  protected String getServerName() {
+    return "malformedurl-test";
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    setUpServer();
+    registerHandler(GET, WILDCARD.substring(1), "Success!");
```

Comment:
change the constant instead of doing a substring, I don't think it's used elsewhere